torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1259,6 +1260,41 @@ def generator():
 
     return list(generator())
 
+def sample_inputs_batch_norm(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+    make_arg_without_requires_grad = partial(make_tensor, device=device, dtype=dtype, requires_grad=False)
+
+    # Ordered as: input shape, kwargs for training, momentum, eps
+    cases: Tuple[Tuple[int], dict] = (  # type: ignore[assignment]
+        ((S, S, S), {'training': True, 'momentum': 0.5, 'eps': 0.6}),
+        ((3, 2, 4), {'training': False, 'momentum': -1.2}),
+        ((3, 1), {'training': True, 'momentum': 0.0}),
+        ((2, 1), {'training': False}),
+        ((3, 2, 3, 4), {'training': True, 'momentum': -1.0, 'eps': 0.5}),
+        ((3, 2, 3, 4), {'training': False, 'momentum': -1.0, 'eps': 0.5}),
+        ((2, 1), {})
```

Comment:
Does batch_norm support a batch dimension of size 0? (I'm not sure, so asking here)

Reply:
Nice point, thanks for asking! Looks like it does:

```python
In [5]: torch.nn.functional.batch_norm(torch.randn(0,), None, None, training=True)
Out[5]: tensor([])

In [5]: torch.nn.functional.batch_norm(torch.randn(0,), None, None, training=True)
Out[5]: tensor([])
```

Will add a case for 0d tensor.

Reply:
For 0-d tensors, with `weight` and `bias` as non-None values, it fails with following error:

```python
In [3]: torch.nn.functional.batch_norm(x, x, x, weight=None, bias=x, training=True)
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-3-dfc6402a839f> in <module>
----> 1 torch.nn.functional.batch_norm(x, x, x, weight=None, bias=x, training=True)

~/anaconda3/lib/python3.8/site-packages/torch/nn/functional.py in batch_norm(input, running_mean, running_var, weight, bias, training, momentum, eps)
   2279         _verify_batch_size(input.size())
   2280
-> 2281     return torch.batch_norm(
   2282         input, weight, bias, running_mean, running_var, training, momentum, eps, torch.backends.cudnn.enabled
   2283     )

IndexError: select(): index 0 out of range for tensor of size [0] at dimension 0
```